ecommerce/extensions/basket/views.py

```
@@ -107,27 +108,13 @@ def get(self, request):
             msg = _('Product [{product}] not available to buy.').format(product=product.title)
             return HttpResponseBadRequest(msg)
 
-        # If the product is not an Enrollment Code and this is a Coupon Redemption request,
-        # we check to see if the user is already enrolled
-        # to prevent double-enrollment and/or accidental coupon usage.
-        if not product.is_enrollment_code_product and code:
-            try:
-                if request.user.is_user_already_enrolled(request, product):
-                    logger.warning(
-                        'User [%s] attempted to repurchase the [%s] seat of course [%s]',
-                        request.user.username,
-                        mode_for_seat(product),
-                        product.attr.course_key
-                    )
-                    msg = _('You are already enrolled in {course}.').format(course=product.course.name)
-                    return HttpResponseBadRequest(msg)
-            except (ConnectionError, SlumberBaseException, Timeout):
-                msg = _('An error occurred while retrieving enrollment details. Please try again.')
-                return HttpResponseBadRequest(msg)
-
         # At this point we're either adding an Enrollment Code product to the basket,
         # or the user is adding a Seat product for which they are not already enrolled
-        prepare_basket(request, [product], voucher)
+        basket = prepare_basket(request, [product], voucher)
+        # If user has already purchased the product basket will be None
+        if not basket:
+            msg = _('You have already purchased this {course} seat.').format(course=product.course.name)
```

@Ayub-Khan What if the basket is `None` because of another error or condition that is different from `You already purchased`? Seems to me you are making the assumption that `None` basket means `already ordered`.
I am afraid this assumption is too implicit to understand.
Can we throw an exception in the `prepare_basket` function when the order is already placed, and then handle the exception here? I feel like Exception throwing and catching is the more explicit way of handling